openmeetings-core/src/test/java/org/apache/openmeetings/core/remote/TestRecordingFlowMocked.java

```diff
@@ -0,0 +1,222 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.openmeetings.core.remote;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Locale;
+import java.util.logging.StreamHandler;
+
+import org.apache.openmeetings.db.dao.label.LabelDao;
+import org.apache.openmeetings.db.dao.record.RecordingDao;
+import org.apache.openmeetings.db.dao.room.RoomDao;
+import org.apache.openmeetings.db.dao.user.UserDao;
+import org.apache.openmeetings.db.entity.basic.Client;
+import org.apache.openmeetings.db.entity.basic.Client.Activity;
+import org.apache.openmeetings.db.entity.basic.Client.StreamDesc;
+import org.apache.openmeetings.db.entity.label.OmLanguage;
+import org.apache.openmeetings.db.entity.record.Recording;
+import org.apache.openmeetings.db.entity.room.Room;
+import org.apache.openmeetings.db.entity.user.User;
+import org.apache.openmeetings.db.manager.IClientManager;
+import org.junit.Test;
+import org.kurento.client.MediaPipeline;
+import org.kurento.client.Transaction;
+import org.mockito.BDDMockito;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+
+import com.github.openjson.JSONObject;
+
+@PrepareForTest(LabelDao.class)
+public class TestRecordingFlowMocked extends BaseMockedTest {
+	private static final Long USER_ID = 1L;
+	private static final Long ROOM_ID = 5L;
+	@Mock
+	private UserDao userDao;
+	@Mock
+	private RoomDao roomDao;
+	@Mock
+	private RecordingDao recDao;
+	@Mock
+	private IClientManager cm;
+	
+	//This variable holds a reference to the current client in the room
+	private Client c;
+	
+	//This variable hold a reference to the UID of the StreamDesc that will be created 
+	private String streamDescUID;
+
+	@Override
+	public void setup() {
+		super.setup();
+		when(client.createMediaPipeline(any(Transaction.class))).thenReturn(mock(MediaPipeline.class));
+		User u = new User();
+		u.setId(USER_ID);
+		u.setFirstname("firstname");
+		u.setLastname("lastname");
+		when(userDao.get(USER_ID)).thenReturn(u);
+		doReturn(true).when(handler).isConnected();
+		when(recDao.update(any(Recording.class))).thenAnswer(new Answer<Recording>() {
+			@Override
+			public Recording answer(InvocationOnMock invocation) throws Throwable {
+				Object[] args = invocation.getArguments();
+				Recording r = (Recording) args[0];
+				r.setId(1L);
+				return r;
+			}
+		});
+		
+		PowerMockito.mockStatic(LabelDao.class);
```

Comment:
this should be moved to `@BeforeAll`

Reply:
I think I did that in my last commit. Let me check